Containerfile.zfs

```diff
@@ -0,0 +1,42 @@
+###
+### Containerfile.common - used to build ONLY NON-nvidia kmods
+###
+
+ARG FEDORA_MAJOR_VERSION="${FEDORA_MAJOR_VERSION:-40}"
+ARG KERNEL_FLAVOR="${KERNEL_FLAVOR:-coreos-stable}"
+ARG KERNEL_IMAGE="${KERNEL_IMAGE:-${KERNEL_FLAVOR}-kernel}"
+ARG KERNEL_ORG="${KERNEL_ORG:-ublue-os}"
+ARG KERNEL_BASE="ghcr.io/${KERNEL_ORG}/${KERNEL_IMAGE}:${FEDORA_MAJOR_VERSION}"
+ARG BUILDER_IMAGE="${BUILDER_IMAGE:-quay.io/fedora/fedora}"
+ARG BUILDER_BASE="${BUILDER_IMAGE}:${FEDORA_MAJOR_VERSION}"
+FROM ${KERNEL_BASE} AS kernel_cache
+FROM ${BUILDER_BASE} AS builder
+
+ARG FEDORA_MAJOR_VERSION="${FEDORA_MAJOR_VERSION:-40}"
+ARG KERNEL_FLAVOR="${KERNEL_FLAVOR:-coreos-stable}"
+ARG RPMFUSION_MIRROR=""
+ARG ZFS_MINOR_VERSION="${ZFS_MINOR_VERSION:-2.2}"
+
+COPY build*.sh /tmp
+COPY certs /tmp/certs
+
+# cached kernel rpms
+COPY --from=kernel_cache /tmp/rpms /tmp/kernel_cache
+
+# Set kernel name
+RUN if grep -q "coreos" <<< "${KERNEL_FLAVOR}"; then \
```

Comment:
Since this is now split out. Should we still have this conditional? We instead are controlling if it runs with the matrix.

Reply:
My only thought was to prevent failures of builds where people attempt it on too new kernels, but... not really required... i'm comfortable with the matrix gate.